chimney/src/main/scala/io/scalaland/chimney/internal/compiletime/derivation/transformer/Gateway.scala

```diff
@@ -99,18 +107,18 @@ private[compiletime] trait Gateway { this: Definitions & Derivation =>
       .flatMap {
         case DerivedExpr.TotalExpr(expr) =>
           ctx match {
-            case TransformerContext.ForTotal(_, _, _, _) =>
+            case TransformerContext.ForTotal(_, _, _, _, _) =>
```

Comment:
I'd stick to type-based matching in such cases (`case _: TransformerContext.ForTotal`).